src/ExceptionConverter.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\FtpExtractor;
+
+use Keboola\Component\UserException;
+use League\Flysystem\FileNotFoundException;
+
+final class ExceptionConverter
+{
+    public static function resolve(\Throwable $e): void
+    {
+        if (self::isUserException($e)) {
+            self::toUser($e);
+        }
+
+        throw $e;
```

Comment:
Napadá mě, že tímhle throwem se ztratí původní stacktrace, ne? Myslím, že to máme i někde jinde, ale říkám si, že by asi bylo lepší ji zabalit do aplikační výjimky jako previous...

Reply:
Pravda, zabalím to do `ApplicationException`